core/src/main/scala/Graph.scala

```diff
@@ -365,20 +364,33 @@ case class Graph[N,A,B](rep: GraphRep[N,A,B]) {
   def edges: Vector[Edge[N]] = labEdges.map { case LEdge(v, w, _) => Edge(v, w) }
 
   /**
-   * Get all the contexts in the graph, as a vector
+   * Get all the contexts in the graph, as a vector. Note that the resulting contexts may
+   * overlap, in the sense that successive contexts in the result will contain vertices
+   * from previous contexts as well.
    * @group projection
    */
   def contexts: Vector[Context[N,A,B]] =
-    fold(Vector[Context[N,A,B]]())(_ +: _)
+    rep.map { case (v, c) => c.toContext(v) }.toVector
```

Comment:
I've changed `contexts` to actually return all the contexts of the graph, rather than successive "tails" of the graph.